checkers_bot_tournament/game.py

```diff
@@ -78,21 +86,55 @@ def import_pdn(self, filename: str) -> None:
 
             move_obj = Move(start_pos, end_pos, removed_pos)
 
-            if not self.board.is_valid_move(self.current_turn, move_obj):
-                raise RuntimeError(f"Invalid move in import_pdn: {move}")
-
-            capture, promotion = self.board.move_piece(move_obj)
-            if capture or promotion:
-                # Reset action move, since capture or promotion occured
-                self.last_action_move = self.move_number
-                if capture:
-                    self._record_capture()
-                if promotion:
-                    self._record_promotion()
+            # Check if the first move is valid for white or black and set
+            # current_turn accordingly
+            if idx == 0:
+                assert not self.move_history
+
+                if self.board.is_valid_move(Colour.WHITE, move_obj):
+                    self.current_turn = Colour.WHITE
+                elif self.board.is_valid_move(Colour.BLACK, move_obj):
+                    self.current_turn = Colour.BLACK
+                else:
+                    raise RuntimeError(
+                        f"First move in import_pdn is invalid for both white and black: {move}"
+                    )
+            else:
+                if not self.board.is_valid_move(self.current_turn, move_obj):
+                    raise RuntimeError(
+                        f"Invalid move in import_pdn for colour: {str(self.current_turn)}, turn: {self.move_number + 1}, move: {move}"
```

Comment:
What if there's a draw by repetition or anyhow a result in the middle of the PDN but it keeps going?

Ok I think the RuntimeError in line 103 should cover White/Black wins. Up to you if you wanna check for draws or leave it as-is cos it's an obscure case. In case you wanted to, you could yoink even more of the logic from make_move().